File: mobile/ios/app/App/App/WidgetSyncPlugin.swift
// ============================================================================
// Source: mobile/ios/app/App/App/WidgetSyncPlugin.swift
// Version: 0.1.0 — 2026-09-11
// Why: The page's one door to the iPhone widgets — the twin of the Android
//      WidgetSyncPlugin, with the same JS name, so src/lib/widget-sync.ts works
//      unchanged. setView({religious, state, world, avestan}) writes the four
//      settings into the App Group and asks WidgetKit to redraw. Nothing else
//      crosses: not the memorial switch, not personal dates. A missing field
//      keeps the site's default.
// Env / Deps: Capacitor 8, WidgetKit; App Group group.im.taghv.app.
// ============================================================================

import Capacitor
import Foundation
import WidgetKit

@objc(WidgetSyncPlugin)
public class WidgetSyncPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "WidgetSyncPlugin"
    public let jsName = "WidgetSync"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "setView", returnType: CAPPluginReturnPromise),
    ]

    @objc func setView(_ call: CAPPluginCall) {
        guard let defaults = UserDefaults(suiteName: "group.im.taghv.app") else {
            call.reject("App Group is not available")
            return
        }
        // The site's defaults, «پیش‌فرض ایران عزیز»: religious and state off, world on.
        defaults.set(call.getBool("religious") ?? false, forKey: "religious")
        defaults.set(call.getBool("state") ?? false, forKey: "state")
        defaults.set(call.getBool("world") ?? true, forKey: "world")
        defaults.set(call.getBool("avestan") ?? false, forKey: "avestan")
        WidgetCenter.shared.reloadAllTimelines()
        call.resolve()
    }
}

File: mobile/android/app/app/src/main/java/im/taghv/app/widget/WidgetSyncPlugin.kt
// ============================================================================
// Source: mobile/android/app/app/src/main/java/im/taghv/app/widget/WidgetSyncPlugin.kt
// Version: 0.1.0 — 2026-09-10
// Why: The one door from the page to the widgets. The page calls
//      WidgetSync.setView({religious, state, world, avestan}) — src/lib/
//      widget-sync.ts — and the widgets redraw with the same groups and month
//      names the calendar on screen uses. Nothing else crosses: not the
//      memorial switch, not personal dates. A missing field keeps the default.
// Env / Deps: Capacitor 8 plugin API; registered in MainActivity.
// ============================================================================

package im.taghv.app.widget

import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin
import im.taghv.core.EventGroups

@CapacitorPlugin(name = "WidgetSync")
class WidgetSyncPlugin : Plugin() {
    @PluginMethod
    fun setView(call: PluginCall) {
        val d = WidgetSettings.DEFAULT
        val settings = WidgetSettings(
            EventGroups(
                religious = call.getBoolean("religious", d.groups.religious) ?: d.groups.religious,
                state = call.getBoolean("state", d.groups.state) ?: d.groups.state,
                world = call.getBoolean("world", d.groups.world) ?: d.groups.world,
            ),
            older = call.getBoolean("avestan", d.older) ?: d.older,
        )
        WidgetSettings.write(context, settings)
        TaghvimWidget.refreshAll(context)
        call.resolve()
    }
}
